runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java

```diff
@@ -191,21 +196,52 @@ private Ellipse2D drawEllipse(Graphics2D graphics, int x, int y)
 
 	private void drawSkillImage(Graphics2D graphics, XpGlobe xpGlobe, int x, int y)
 	{
-		BufferedImage skillImage = iconManager.getSkillImage(xpGlobe.getSkill());
-
-		if (skillImage == null)
+		BufferedImage toDraw;
+		if (xpGlobe.getSkillIcon() != null && xpGlobe.getGlobeSize() == config.xpOrbSize())
 		{
-			return;
+			toDraw = xpGlobe.getSkillIcon();
+		}
+		else
+		{
+			toDraw = generateScaledIcon(xpGlobe.getSkill());
```

Comment:
This doesn't need to be done on every frame.

Reply:
Currently this else block will only execute if ```xpGlobe``` is not set to anything, or if the globe size has recently been changed (requires scaling again).

After which, the scaled image is saved in the xpGlobe instance, so the next time we need to draw the icon, it will just set ```toDraw = xpGlobe.getSkillIcon()```, rather than generating a new one.